test/types/index.ts

```diff
@@ -0,0 +1,132 @@
+import agent from '../../'
+
+agent.start({
+  captureExceptions: false,
+  logLevel: 'fatal'
+})
+
+function started (bool: boolean) {}
+started(agent.isStarted())
+
+agent.lambda(() => {})
+agent.lambda('foo', () => {})
+
+agent.handleUncaughtExceptions((err: Error) => {
+  console.error(err.stack)
+  process.exit(1)
+})
+
+agent.captureError(new Error('foo'))
+agent.captureError('foo')
+agent.captureError({ message: 'hello %s', params: ['world'] })
+agent.captureError(new Error('foo'), { tags: { foo: 'bar' } })
+agent.captureError('foo', { tags: { foo: 'bar' } })
+agent.captureError({ message: 'hello %s', params: ['world'] }, { tags: { foo: 'bar' } })
+agent.captureError(new Error('foo'), { tags: { foo: 'bar' } }, () => {})
+agent.captureError('foo', { tags: { foo: 'bar' } }, () => {})
+agent.captureError({ message: 'hello %s', params: ['world'] }, { tags: { foo: 'bar' } }, () => {})
+agent.captureError(new Error('foo'), () => {})
+agent.captureError('foo', () => {})
+agent.captureError({ message: 'hello %s', params: ['world'] }, () => {})
+
+agent.startTransaction()
+agent.startTransaction('foo')
+agent.startTransaction('foo', 'bar')
+agent.startTransaction('foo', 'bar', { startTime: 1 })
+
+agent.setTransactionName('foo')
+
+agent.endTransaction()
+
+agent.startSpan()
+agent.startSpan('foo')
+agent.startSpan('foo', 'bar')
+agent.startSpan('foo', 'bar', { childOf: 'baz' })
+
+agent.setTag('foo', 'bar')
+agent.setTag('foo', 1)
+agent.setTag('foo', false)
+
+agent.addTags({ s: 'bar', n: 42, b: false })
+
+agent.setUserContext({
+  id: 'foo',
+  username: 'bar',
+  email: 'baz'
+})
+
+agent.setCustomContext({ foo: { bar: { baz: true }}})
+
+function filter1 (payload: object) {
+  return payload
+}
+function filter2 (payload: object) {
+  return false
+}
+function filter3 (payload: object) {}
+agent.addFilter(filter1)
+agent.addFilter(filter2)
```

Comment:
@delvedor @sqren This test fails because it detects that `filter2` returns a boolean, but the `addFilter` function expects the return value to be `false`. Obviously in this case, that boolean value is `false`, but it's somehow not allowing it. Any idea how I can update the types to allow a falsy value here in a way that would allow the user to return false inside the `FilterFn`?

Reply:
Here's the VS Code error:

![image](https://user-images.githubusercontent.com/10602/54684986-f945ac00-4b15-11e9-81b3-f7a2b026e230.png)

Reply:
Typescript is expanding the type of `false` to boolean. You could narrow it explicitly:

```ts
function filter2(payload: object): false {
  return false;
}
```

But that's pretty ugly. Another approach is something like this:
```ts
const filter4: Filter = (payload: object) => {
  return false;
};
```

A bit better but still not great.

Reply:
Hmm I see. Maybe I should change the API to be more TypeScript friendly then

Reply:
On the other hand that would be a breaking change, so I can't do that in this iteration 😞🤬

Reply:
Fixed